src/operators.jl

```diff
@@ -236,20 +236,119 @@ julia> collect(edges(f))
  Edge 2 => 3
 ```
 """
-function symmetric_difference(g::T, h::T) where T <: AbstractGraph
-    gnv = nv(g)
-    hnv = nv(h)
+function symmetric_difference(g::T, h::T) where T <: SimpleDiGraph
```

Comment:
Oh, wait. This should be:
```
@traitfn function symmetric_difference(g::::IsDirected, h::::IsDirected)
```

and the one for undirected graphs should be `(!IsDirected)`.

Reply:
I forgot that that syntax doesn't work, but I think you got the correct one.